completion/services.py

```diff
@@ -76,6 +76,44 @@ def fill_in_run(block_key):
                 completions[candidate] = 0.0
         return completions
 
+    def get_selected_library_children(self, library_node):
+        """
+        Content Libraries will select a subset of the content in their library to
+        show to each learner. This function takes in a library content node and
+        returns the subset of its children that were selected for this user.
+        """
+        selected_children = []
+        all_children = library_node.get_children()
+        for block_type, block_id in library_node.selected:
+            selected_key = library_node.location.course_key.make_usage_key(block_type, block_id)
+            selected_children.extend(
+                list(filter(
+                    lambda child, selected_key=selected_key: selected_key == child.scope_ids.usage_id, all_children
```

Comment:
You're passing in a new keyword argument to a lambda? Not sure I've seen that before, interesting. But couldn't you just refer to `selected_key` as in-scope even without the keyword arg?

Reply:
Yeah, I was getting some pylint error about how selected_key changes value and how it's just taking the implied value from the scope. Googling around led me to passing it into the lambda as a kwarg and it fixed the pylint issue.